File: Othello-Src-Cpp/src/board.cpp
#include <board.h>

/*
    Constructor
*/
Board::Board() { this->clear(); }

/*
    Operator ==
*/
bool Board::operator==(const Board &rhs) { return (this->board == rhs.board); }

/*
    Operator =
*/
void Board::operator=(const Board &rhs) { this->board = rhs.board; }

/*
    Get Status of the Cell of the Board
*/
Status Board::get(int _x, int _y) {
  if (this->check_boundary(_x, _y))
    return this->board.at(_x).at(_y);
  else
    return Status::ILLEGAL;
}

/*
    Set Status of the Cell of the Board
*/
bool Board::set(int _x, int _y, Status _status) {
  if (this->check_boundary(_x, _y)) {
    this->board.at(_x).at(_y) = _status;
    return true;
  } else {
    return false;
  }
}

/*
    Clear the Board
*/
void Board::clear() {
  for (uint i = 0; i < BOARD_H; i++)
    for (uint j = 0; j < BOARD_W; j++)
      this->board.at(i).at(j) = Status::UNOCCUPIED;

  this->board.at(0).at(0)                     = Status::ILLEGAL;
  this->board.at(0).at(BOARD_W - 1)           = Status::ILLEGAL;
  this->board.at(BOARD_H - 1).at(0)           = Status::ILLEGAL;
  this->board.at(BOARD_H - 1).at(BOARD_W - 1) = Status::ILLEGAL;
}

/*
    Check whether the coordinate is inside board or not
*/
bool Board::check_boundary(int _x, int _y) {
  return (0 <= _x) && (_x < BOARD_H) && (0 <= _y) && (_y < BOARD_W);
}

File: Othello-Src-Python/board.py
#!/usr/bin/env python3
from enum import Enum
from copy import deepcopy

# Board Size
BOARD_H = 8
BOARD_W = 8
board_size = (BOARD_H, BOARD_W)


class Status(Enum):
    # Board Status of Cell
    UNOCCUPIED = 0
    BLACK = 1
    WHITE = 2
    ILLEGAL = 3


class Board():
    """ Class Board """

    def __init__(self):
        self.board = []
        self.clear()

    def __repr__(self):
        return self.__str__()

    def __str__(self):
        msg = ""
        for r in range(BOARD_H):
            for c in range(BOARD_W):
                msg += str(self.board[r][c].value) + ' '
            msg += '\n'
        return msg

    def __eq__(self, rhs):
        return self.board == rhs.board

    def clear(self):
        for r in range(BOARD_H):
            self.board.append([])
            for _ in range(BOARD_W):
                self.board[r].append(Status.UNOCCUPIED)

        self.board[0][0] = Status.ILLEGAL
        self.board[0][BOARD_W-1] = Status.ILLEGAL
        self.board[BOARD_H-1][0] = Status.ILLEGAL
        self.board[BOARD_H-1][BOARD_W-1] = Status.ILLEGAL

    def get(self, _x, _y):
        if(self.check_boundary(_x, _y)):
            return self.board[_x][_y] 
        else:
            return Status.ILLEGAL
            
    def set(self, _x, _y, _value):
        if(self.check_boundary(_x, _y)):
            if(type(_value) == int):
                self.board[_x][_y] = Status(_value)
            elif(type(_value) == type(Status.UNOCCUPIED)):
                self.board[_x][_y] = _value

    def check_boundary(self, _x, _y):
        return (0 <= _x) and (_x < BOARD_H) and (0 <= _y) and (_y < BOARD_W)
